components/gh_events/GithubFeedFilter.tsx

```diff
@@ -0,0 +1,59 @@
+"use client";
+import { FilterOption } from "@/lib/types";
+import { usePathname, useRouter, useSearchParams } from "next/navigation";
+
+const GithubFeed = (props: { filterEvents: FilterOption[] }) => {
+  const { filterEvents } = props;
+  const router = useRouter();
+  const pathname = usePathname();
+  const searchParams = useSearchParams();
+
+  const updateSearchParam = (key: string, value: string) => {
+    const params = new URLSearchParams(searchParams.toString());
+
+    value !== "All" ? params.set(key, value) : params.delete(key);
```

Comment:
Use null/undefined instead of "All" for all.
```suggestion
    !!value ? params.set(key, value) : params.delete(key);
```